src/cli/runtime/http.rs

```diff
@@ -182,8 +183,41 @@ impl HttpIO for NativeHttp {
         )
         .await?)
     }
+
+    fn cl(&self) -> Box<dyn HttpIO> {
+        Box::new(self.clone())
```

Comment:
Cloning isn't going to help, ideally we want to create separate connection pools and caches for each http-client.

Reply:
I have a question about this. According to `https://docs.rs/reqwest/0.12.5/reqwest/struct.Client.html`, `reqwest::Client` has a connection pool in it already. Do we need more work?

Reply:
Nothing special needs to be done for pooling.